### PR TITLE
Add vim-matchup

### PR DESCRIPTION
**Pull request by kiryph**

### CODE REVIEW

File: src/lib/manual.json [hunk 1]
```diff
@@ -78,6 +78,18 @@
...
+    {
+      "type": "github",
+      "username": "andymass",
+      "repo": "vim-matchup",
+      "tags": [
+        "keybinding",
+        "programming-languages-support",
+        "motion",
+        "treesitter-based",
+        "text-objects"
```
Comment (neurosnap) [inline src/lib/manual.json:91]: let's not create a new tag

Comment (kiryph) [reply src/lib/manual.json:91]: I like plugins which provide new text objects and I think there are quite a few and the name of a plugin does not gives this away. Are you strongly against the tag `text-objects`?

For vim there is following website which lists many possible ones: https://github.com/kana/vim-textobj-user/wiki

Comment (neurosnap) [reply src/lib/manual.json:91]: I'm not against it but most of these tags are derived from https://github.com/rockerBOO/awesome-neovim so modifying them might cause conflicts with other names.

**Comment by neurosnap** [comment]
Closing for now.  Feel free to reopen with conflicts resolved

**Pull request closed without merging**